Assets/Scripts/Models/Buildable/Utility.cs

```diff
@@ -257,21 +269,37 @@ public static Utility PlaceInstance(Utility proto, Tile tile)
             for (int ypos = y - 1; ypos < y + 2; ypos++)
             {
                 Tile tileAt = World.Current.GetTileAt(xpos, ypos, tile.Z);
-                if (tileAt != null && tileAt.Utilities != null)
+                if (tileAt != null && tileAt.Utilities != null && tileAt.Utilities.ContainsKey(obj.Name))
                 {
-                    foreach (Utility utility in tileAt.Utilities.Values)
+                    Utility utility = tileAt.Utilities[obj.Name];
+                    if (utility.Changed != null)
                     {
-                        if (utility.Changed != null)
-                        {
-                            utility.Changed(utility);
-                        }
+                        utility.Changed(utility);
                     }
                 }
             }
         }
 
+        if (!skipGridUpdate)
+        {
+            obj.UpdateGrid(obj);
+        }
+        else
+        {
+            // If we're skipping the update, we need a temporary grid for furniture in the same tile to connect to.
+            obj.Grid = new Grid();
+            World.Current.PowerNetwork.RegisterGrid(obj.Grid);
+        }
+
+        if (obj.Tile != null && obj.Tile.Furniture != null && obj.Tile.Furniture.GetComponent<PowerConnection>("PowerConnection") != null)
```

Comment:
maybe create IEnumerable<IPlugable> GetPlugableComponents() on Furniture class and use it here